iptables/iptables.go

```diff
@@ -330,12 +357,19 @@ func Exists(table Table, chain string, rule ...string) bool {
 	if supportsCOpt {
 		// if exit status is 0 then return true, the rule exists
 		_, err := Raw(append([]string{"-t", string(table), "-C", chain}, rule...)...)
+		if err != nil {
+			_, err = Raw(append([]string{"-t", string(table), "-C", chain}, append(rule, commentArgs...)...)...)
+		}
 		return err == nil
 	}
 
 	// parse "iptables -S" for the rule (it checks rules in a specific chain
 	// in a specific table and it is very unreliable)
-	return existsRaw(table, chain, rule...)
+	if existsRaw(table, chain, rule...) {
```

Comment:
Yes, the `rawExists()` check works only for very simple rules (where the `-m` does not get added, for ex.). This is why I urged to fix the -C option handling in the iptables.Exists() function, so that we now never fall back to use the `rawExists()` which is unreliable (provided user has at least 1.4.11 ver).

But @chenchun brings up a good point that now with the comment addition `rawExists()` will fail also for simple rules, given the comment position in the iptables -S o/p.

Reason why I like the comment addition is because it will allow us to cleanup any left over rule from default chains which may be there because of an ungraceful shutdown. We can detect them just looking for the comment.